--- Pound-2.7/svc.c~	2015-04-29 19:47:57.000000000 +0300
+++ Pound-2.7/svc.c	2015-04-29 19:48:28.151076743 +0300
@@ -1563,55 +1563,55 @@
             break;
         case CTRL_EN_LSTN:
             if((lstn = sel_lstn(&cmd)) == NULL)
-                logmsg(LOG_INFO, "thr_control() bad listener %d", cmd.listener);
+                logmsg(LOG_NOTICE, "thr_control() bad listener %d", cmd.listener);
             else
                 lstn->disabled = 0;
             break;
         case CTRL_DE_LSTN:
             if((lstn = sel_lstn(&cmd)) == NULL)
-                logmsg(LOG_INFO, "thr_control() bad listener %d", cmd.listener);
+                logmsg(LOG_NOTICE, "thr_control() bad listener %d", cmd.listener);
             else
                 lstn->disabled = 1;
             break;
         case CTRL_EN_SVC:
             if((svc = sel_svc(&cmd)) == NULL)
-                logmsg(LOG_INFO, "thr_control() bad service %d/%d", cmd.listener, cmd.service);
+                logmsg(LOG_NOTICE, "thr_control() bad service %d/%d", cmd.listener, cmd.service);
             else
                 svc->disabled = 0;
             break;
         case CTRL_DE_SVC:
             if((svc = sel_svc(&cmd)) == NULL)
-                logmsg(LOG_INFO, "thr_control() bad service %d/%d", cmd.listener, cmd.service);
+                logmsg(LOG_NOTICE, "thr_control() bad service %d/%d", cmd.listener, cmd.service);
             else
                 svc->disabled = 1;
             break;
         case CTRL_EN_BE:
             if((svc = sel_svc(&cmd)) == NULL) {
-                logmsg(LOG_INFO, "thr_control() bad service %d/%d", cmd.listener, cmd.service);
+                logmsg(LOG_NOTICE, "thr_control() bad service %d/%d", cmd.listener, cmd.service);
                 break;
             }
             if((be = sel_be(&cmd)) == NULL)
-                logmsg(LOG_INFO, "thr_control() bad backend %d/%d/%d", cmd.listener, cmd.service, cmd.backend);
+                logmsg(LOG_NOTICE, "thr_control() bad backend %d/%d/%d", cmd.listener, cmd.service, cmd.backend);
             else
                 kill_be(svc, be, BE_ENABLE);
             break;
         case CTRL_DE_BE:
             if((svc = sel_svc(&cmd)) == NULL) {
-                logmsg(LOG_INFO, "thr_control() bad service %d/%d", cmd.listener, cmd.service);
+                logmsg(LOG_NOTICE, "thr_control() bad service %d/%d", cmd.listener, cmd.service);
                 break;
             }
             if((be = sel_be(&cmd)) == NULL)
-                logmsg(LOG_INFO, "thr_control() bad backend %d/%d/%d", cmd.listener, cmd.service, cmd.backend);
+                logmsg(LOG_NOTICE, "thr_control() bad backend %d/%d/%d", cmd.listener, cmd.service, cmd.backend);
             else
                 kill_be(svc, be, BE_DISABLE);
             break;
         case CTRL_ADD_SESS:
             if((svc = sel_svc(&cmd)) == NULL) {
-                logmsg(LOG_INFO, "thr_control() bad service %d/%d", cmd.listener, cmd.service);
+                logmsg(LOG_NOTICE, "thr_control() bad service %d/%d", cmd.listener, cmd.service);
                 break;
             }
             if((be = sel_be(&cmd)) == NULL) {
-                logmsg(LOG_INFO, "thr_control() bad back-end %d/%d", cmd.listener, cmd.service);
+                logmsg(LOG_NOTICE, "thr_control() bad back-end %d/%d", cmd.listener, cmd.service);
                 break;
             }
             if(ret_val = pthread_mutex_lock(&svc->mut))
@@ -1622,7 +1622,7 @@
             break;
         case CTRL_DEL_SESS:
             if((svc = sel_svc(&cmd)) == NULL) {
-                logmsg(LOG_INFO, "thr_control() bad service %d/%d", cmd.listener, cmd.service);
+                logmsg(LOG_NOTICE, "thr_control() bad service %d/%d", cmd.listener, cmd.service);
                 break;
             }
             if(ret_val = pthread_mutex_lock(&svc->mut))
