diff -urNp -x '*.orig' Pound-2.8.org/config.c Pound-2.8/config.c
--- Pound-2.8.org/config.c	2018-05-11 12:16:05.000000000 +0200
+++ Pound-2.8/config.c	2021-10-04 21:55:51.816168060 +0200
@@ -521,7 +521,7 @@ t_hash(const TABNODE *e)
     char            *k;
 
     k = e->key;
-    res = 2166136261;
+    res = 2166136261UL;
     while(*k)
         res = ((res ^ *k++) * 16777619) & 0xFFFFFFFF;
     return res;
diff -urNp -x '*.orig' Pound-2.8.org/svc.c Pound-2.8/svc.c
--- Pound-2.8.org/svc.c	2018-05-11 12:16:05.000000000 +0200
+++ Pound-2.8/svc.c	2021-10-04 21:55:51.812834729 +0200
@@ -553,7 +553,7 @@ hash_backend(BACKEND *be, int abs_pri, c
     BACKEND         *res, *tb;
     int             pri;
 
-    hv = 2166136261;
+    hv = 2166136261UL;
     while(*key)
         hv = ((hv ^ *key++) * 16777619) & 0xFFFFFFFF;
     pri = hv % abs_pri;
