--- Pound-3.0.1/CMakeLists.txt~	2021-08-23 17:31:52.000000000 +0200
+++ Pound-3.0.1/CMakeLists.txt	2021-10-25 22:47:18.511993405 +0200
@@ -46,10 +46,12 @@
 target_link_libraries(pound -lpthread)
 target_link_libraries(pound Threads::Threads)
 
+if(WANT_TCMALLOC)
 FIND_LIBRARY(TCMALLOC tcmalloc )
 if(TCMALLOC)
 target_link_libraries(pound -ltcmalloc)
 endif()
+endif()
 
 FIND_LIBRARY(MBEDCRYPTO mbedcrypto)
 if(MBEDCRYPTO)
